core/src/main/scala/Json.scala

```diff
@@ -245,8 +230,90 @@ object Json {
+    ((GithubDeploymentEventDecoder |||
+     GithubReleaseEventDecoder: DecodeJson[Github.Event]) |||
+     GithubPullRequestEventDecoder: DecodeJson[Github.Event]) |||
+     GithubPingEventDecoder
+
+  /*
+   * {
+   *   "deployment": {
+   *     "url": "https://api.github.com/repos/timperrett/example/deployments/107241174",
+   *     "id": 107241174,
+   *     "node_id": "MDEwOkRlcGxveW1lbnQxMDcyNDExNzQ=",
+   *     "sha": "fdb7da2ab3b2cd172e86c1af9adefa3523f6d65b",
+   *     "ref": "fdb7da2ab3b2cd172e86c1af9adefa3523f6d65b",
+   *     "task": "deploy",
+   *     "payload": "{ \"foo\": true }",
+   *     "original_environment": "production",
+   *     "environment": "production",
+   *     "description": null,
+   *     "creator": {
+   *      ....
+   *     },
+   *     "created_at": "2018-10-06T04:52:40Z",
+   *     "updated_at": "2018-10-06T04:52:40Z",
+   *     "statuses_url": "https://api.github.com/repos/timperrett/example/deployments/107241174/statuses",
+   *     "repository_url": "https://api.github.com/repos/timperrett/example"
+   *   },
+   *   "repository": {
+   *     "id": 140525376,
+   *     "node_id": "MDEwOlJlcG9zaXRvcnkxNDA1MjUzNzY=",
+   *     "name": "example",
+   *     "full_name": "timperrett/example",
+   *     ....
+   *   },
+   *   "sender": {
+   *    ....
+   *   }
+   * }
+  */
+  implicit val GithubDeploymentEventDecoder: DecodeJson[Github.Deployment] =
+    DecodeJson(z => for {
+      a <- (z --\ "deployment" --\ "id").as[Long]
+      x <- (z --\ "repository" --\ "full_name").as[String]
+      b <- Slug.fromString(x).map(DecodeResult.ok
+           ).valueOr(e => DecodeResult.fail(e.getMessage,z.history))
+      c <- (z --\ "deployment" --\ "ref").as[String]
+      d <- (z --\ "deployment" --\ "environment").as[String]
+      e <- (z --\ "deployment" --\ "payload").as[String]
+      f <- (z --\ "repository" --\ "id").as[Long]
+      g <- (z --\ "deployment" --\ "url").as[String]
+    } yield {
+      // NOTE(timperrett): this seems a little sketchy as we're invoking the
+      // protobuf decoder right here in the JSON decoder, even thought we've
+      // no idea if things might work here or not... we can do better.
+      val bytes = java.util.Base64.getDecoder.decode(e)
+      val unmarshalled = nelson.api.deployable.Deployables.parseFrom(bytes)
+      val converted = unmarshalled.deployables.toList.map { a =>
+        val v = a.version.get
+        Manifest.Deployable(
+          name = a.unitName,
+          version = Version(v.series, v.feature, v.patch),
+          output = Manifest.Deployable.Container(a.kind.container.get.image)
+        )
+      }
+      Github.Deployment(
+        id = a,
+        slug = b,
+        ref = c,
+        environment = d,
+        repositoryId = f,
+        deployables = converted,
+        url = g
+      )
+    })
+
+  // TODO(timperrett): what do we do here about encoding the assets that
+  // are shipped to us as proto format?
```

Comment:
If proto is the only format we expect to be sent then I'd say encode it like we received it.